### DISCUSSION TITLE
Script Club Stat

### DISCUSSION THREAD

**Discussion by ThomasGriat** [JeromeMathieuEcology/club-iEES#24] [Learning ressources]
On pourra trouver dans cette discussion des scripts divers potentiellement utiles à tous qui ont été précédemment vus dans le club stat.

**Comment by ThomasGriat** [comment]
**SCRIPT AFFICHAGE D'UN MODÈLE GAM**

Voila quelques lignes de code vu au club stat du 08/06 sur une façon d'afficher un modèle gam comprenant une structuration des données. 


#Quelques info 
#https://www.r-bloggers.com/2021/03/how-to-solve-common-problems-with-gams/
#https://www.rdocumentation.org/packages/robustgam/versions/0.1.7/topics/robustgam
#https://r.qcbs.ca/workshop08/pres-fr/workshop08-pres-fr.html#147


library(mgcv)
library(plotmod)
library(visreg)

#Donnée avec les replicats en ligne. 

 1 colonne avec x en numerique 
 1 colonne avec y en numerique 
 plusieurs colonnes avec des facteurs
Ici ,  y=n,  x=mois_nbr,   1 facteur = HABITAT

model1 <- gam(n ~ s(mois_nbr,k=5, by = HABITAT), data= inflo_3) 


summary(model1)

#Affichage de base
par(mfrow=c(1,2))
visreg(model1)
par(mfrow=c(1,1))

#Affichage du gam avec un plot / Facteur 
visreg(model1, xvar = "mois_nbr",
       by = "HABITAT", data = inflo_3,
       method = "REML")


#Affichage du gam en 3D (le facteur prends la 3e dimention du plot)
vis.gam(model1,ticktype="detailed",plot.type="contour")

vis.gam(model1,ticktype="detailed",plot.type="persp")

vis.gam(model1, theta = 120, n.grid = 50, lwd = .4)

**Reply by JeromeMathieuEcology** [reply]
en fait, renseignement pris, `s( X1, by = X2)` fait une interaction entre X1 et X2, pas un modèle mixte

pour avoir un effet random ~ X2 il faut utiliser : `s(X2, bs="re")`

Par ex pour avoir un effet interecpt alétoire + un effet 'pente' aléatoire de X2 :
`[...] + s(X2, bs="re") + s(X1,X2, bs="re)`
le 1er s() est pour l'intercept, le 2ème est pour l'effet de X2 sur le smooth

une bonne explication [ici](https://fromthebottomoftheheap.net/2021/02/02/random-effects-in-gams/) sur le blog de Gavin Simpson
